frontend/src/components/design-system/input/ButtonComponent.vue

```diff
@@ -0,0 +1,270 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+import ButtonText from '@/components/design-system/typography/ButtonText.vue';
+
+export interface Props {
+  tag?: 'button' | 'a';
+  href?: string;
+  kind: 'link' | 'outlined' | 'filled' | 'answer';
+  size?: 'medium' | 'small';
+  color?: 'neutral' | 'primary' | 'secondary';
+  selected?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  tag: 'button',
+  href: undefined,
+  size: 'medium',
+  color: 'neutral',
+  selected: false,
+});
+
+const classes = computed(() => ({
+  [`button--${props.kind}`]: props.kind,
+  [`button--${props.size}`]: props.size,
+  [`button--${props.color}`]: props.color,
+  [`button--${props.selected}`]: props.selected,
+}));
+
+const computedTag = computed(() => (props.href ? 'a' : props.tag));
+
+const computedSize = computed(() =>
+  props.kind === 'link' ? props.size : 'medium'
+);
+</script>
+
+<template>
+  <component :is="computedTag" :href="href" :class="['button', classes]">
+    <span
+      v-if="$slots.icon"
+      :class="['icon', $slots.default ? 'icon--before' : 'icon--only']"
+    >
+      <slot name="icon" :size="computedSize" />
+    </span>
+    <ButtonText v-if="$slots.default" :size="computedSize">
+      <slot />
+    </ButtonText>
+    <span v-if="$slots.iconAfter" :class="['icon', 'icon--after']">
+      <slot name="iconAfter" :size="computedSize" />
+    </span>
+  </component>
+</template>
+
+<style scoped lang="scss">
+.button {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  line-height: 0;
+  padding: 0;
+  cursor: pointer;
+  background-color: transparent;
+  text-decoration: none;
+
+  &:disabled {
+    cursor: not-allowed;
```

Comment:
Není úplně obvyklé, že by se třeba na odkazu měnil kurzor. Na odkazu se spíš nastavuje `pointer-events: none`, aby se doopravdy odkaz zneplatnil a nešlo na něj klikat.

Reply:
Jo, pro odkaz ten disabled state nefunguje. Opravím.

Reply:
Ne, vždyť to je blbost, odkaz prostě disabled stav vůbec nemá… Takže to max. ošetřit v props, což teda bude pěkný oříšek…

Reply:
Ano, asi by se to mělo vyřešit přes ty interfacy. U toho použití bych se v zásadě držel takového rule of thumb, že jestli dává smysl otevírat to v novém okně, tak by to měl být odkaz a ne tlačítko.